docs/modules/cloud/device_controller.md

```diff
@@ -0,0 +1,68 @@
+# Device Controller
+
+
+## Device Controller Overview
+ The device controller is the cloud component of KubeEdge which is responsible for device management. Device management im KubeEdge is implemented by making use of Kubernetes
+ [Custom Resource Definitions (CRDs)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to describe device metadata/status and device controller to synchronize these device updates between edge and cloud. 
+ The device controller starts two separate goroutines called `upstream controller` and `downstream controller`. These are not separate controllers as such but named here for clarity.
+
+The device controller makes use of device model and device instance to implement device management :
+ - **Device Model**: A `device model` describes the device properties exposed by the device and property visitors to access these properties. A device model is like a reusable template using which many devices can be created and managed.
```

Comment:
we can add links to api types in the design document.

Reply:
also add samples for device models and instances which are available in the repository.

Reply:
Done, thanks